junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -23,114 +24,79 @@ import org.junit.jupiter.api.function.ThrowingSupplier
  * @see Assertions.fail
  */
 fun fail(message: String?, throwable: Throwable? = null): Nothing =
-    Assertions.fail<Nothing>(message, throwable)
+    Assertions.fail(message, throwable)
 
 /**
  * @see Assertions.fail
  */
 fun fail(message: (() -> String)?): Nothing =
-    Assertions.fail<Nothing>(message)
+    Assertions.fail(message)
 
 /**
  * @see Assertions.fail
  */
 fun fail(throwable: Throwable?): Nothing =
-    Assertions.fail<Nothing>(throwable)
-
-/**
- * [Stream] of functions to be executed.
- */
-private typealias ExecutableStream = Stream<() -> Unit>
-private fun ExecutableStream.convert() = map { Executable(it) }
+    Assertions.fail(throwable)
 
 /**
  * @see Assertions.assertAll
  */
-fun assertAll(executables: ExecutableStream) =
-    Assertions.assertAll(executables.convert())
+fun assertAll(executables: Stream<() -> Unit>) =
+    assertAll(null, executables)
 
 /**
  * @see Assertions.assertAll
  */
-fun assertAll(heading: String?, executables: ExecutableStream) =
-    Assertions.assertAll(heading, executables.convert())
-
-/**
- * [Collection] of functions to be executed.
- */
-private typealias ExecutableCollection = Collection<() -> Unit>
-private fun ExecutableCollection.convert() = map { Executable(it) }
+fun assertAll(heading: String?, executables: Stream<() -> Unit>) =
+    Assertions.assertAll(heading, executables.map { Executable(it) })
 
 /**
  * @see Assertions.assertAll
  */
-fun assertAll(executables: ExecutableCollection) =
-    Assertions.assertAll(executables.convert())
+fun assertAll(executables: Collection<() -> Unit>) =
+    assertAll(executables.stream())
 
 /**
  * @see Assertions.assertAll
  */
-fun assertAll(heading: String?, executables: ExecutableCollection) =
-    Assertions.assertAll(heading, executables.convert())
+fun assertAll(heading: String?, executables: Collection<() -> Unit>) =
+    assertAll(heading, executables.stream())
 
 /**
  * @see Assertions.assertAll
  */
 fun assertAll(vararg executables: () -> Unit) =
-    assertAll(executables.toList().stream())
+    assertAll(Arrays.stream(executables))
 
 /**
  * @see Assertions.assertAll
  */
 fun assertAll(heading: String?, vararg executables: () -> Unit) =
-    assertAll(heading, executables.toList().stream())
+    assertAll(heading, Arrays.stream(executables))
 
 /**
- * Example usage:
- * ```kotlin
- * val exception = assertThrows<IllegalArgumentException> {
- *     throw IllegalArgumentException("Talk to a duck")
- * }
- * assertEquals("Talk to a duck", exception.message)
- * ```
+ * @sample org.junit.jupiter.api.AssertionsSamples.assertThrowsSample
```

Comment:
Does the Kotlin compiler validate these? It can‘t see test classes when compiling these, though, can it?

Reply:
I don't know what happens under the hood but can find out.